Plan bee

muhamadazmy opened the pull request.

muhamadazmy committed:
Initial commit

We regernate the types, next we gonna build a cleaner types
for mongo

muhamadazmy committed:
Generate correct id for root types

muhamadazmy committed:
Implementing the farm api

Still far from complete, but gives an idea
on how things should be organized

muhamadazmy committed:
Add email validation

muhamadazmy committed:
Cleaner action method

muhamadazmy committed:
Work on node endpoints

muhamadazmy committed:
Add proofs

muhamadazmy committed:
Re-organizing code

muhamadazmy committed:
WIP: user pkg implementation

muhamadazmy committed:
Implement user find

muhamadazmy committed:
Finish user package

muhamadazmy committed:
WIP: workload package implementation

So far, done with reservation verification and
customer signature verification

muhamadazmy committed:
Working on reservations

Return workloads on poll

muhamadazmy committed:
Get workload by id

muhamadazmy committed:
WIP: setting result

muhamadazmy committed:
Implement a result put method

muhamadazmy committed:
WIP: some refactoring to avoid cyclic import

muhamadazmy committed:
Full cycle of reservation

muhamadazmy committed:
WIP: handle delete reservation

We need to make sure to delete all reservations in one query so
we don't endup with multiple reservations of the same id

muhamadazmy committed:
Fix workload listing to include to-delete reservations

muhamadazmy committed:
refactor tfuser

muhamadazmy committed:
WIP: tfuser refactoring

muhamadazmy committed:
Make sure reservation that deleted return workloads with to_delete=true

muhamadazmy committed:
Update used capacity

muhamadazmy committed:
Run mod tidy

muhamadazmy committed:
Linting

muhamadazmy committed:
Fix unti-tests

zaibon commented on tools/bcdb_mock/pkg/directory/farms_store.go:
```diff
@@ -0,0 +1,54 @@
+}
+
+// Add add farm to store
+// TODO: support update farm information ?
```
yes indeed, we need to be able to update some of the field of a farm. see: https://github.com/threefoldtech/jumpscaleX_threebot/blob/ab4ddcb7014dd226affe86d5d891e4881e81ed87/ThreeBotPackages/tfgrid/directory/actors/farms.py#L62-L75

zaibon commented on tools/bcdb_mock/pkg/directory/node_handlers.go:
```diff
@@ -0,0 +1,242 @@
+	return nil, nil
+}
+
+func (s *NodeAPI) updateUsedResources(r *http.Request) (interface{}, mw.Response) {
```
This should be `updateReservedResources` we do not track "Used" resources, only total and reserved

zaibon commented on tools/tfuser/cmds_identity.go:
```diff
@@ -24,10 +28,16 @@ func cmdsGenerateID(c *cli.Context) error {
-	fmt.Printf("new identity generated: %s\n", k.Identity())
-	fmt.Printf("seed saved at %s\n", output)
+
+	fmt.Printf("Your ID is: %d\n", id)
```
I think we should also store the user ID somewhere no ?

muhamadazmy committed:
Support siging

muhamadazmy committed:
Apply some review comments

muhamadazmy committed:
Implement work queue for reservations

This queue is filled with workloads that the nodes
has to process even if they are old reservations.
for example, to-delete, or signed reservatinos

muhamadazmy committed:
Make sure to close the cursor

muhamadazmy committed:
Remove debug code

muhamadazmy committed:
WIP: signing for deletion

NOTE: need to be tested

muhamadazmy committed:
Fix tfuser delete and test delete operation

muhamadazmy committed:
Fix tests

muhamadazmy committed:
Support more than one result per workload

Use the node id also to make sure multiple results
are supported. This is only usable with network reservations

muhamadazmy committed:
Fix linting

muhamadazmy committed:
more linting

muhamadazmy committed:
more fixes

muhamadazmy committed:
Fix after a merge
